src/runtime/float.cpp

```diff
@@ -868,27 +868,15 @@ template <ExceptionStyle S> static BoxedFloat* _floatNew(Box* a) noexcept(S == C
 
         return res;
     } else {
-        static BoxedString* float_str = internStringImmortal("__float__");
```

Comment:
Can we maybe just replace this whole __floatNew function implementation with a call to:
```
if (PyString_CheckExact(x))
        return PyFloat_FromString(x, NULL);
    return PyNumber_Float(x);
```
Because I think everything we do here ```PyNumber_Float``` should already do